### PR TITLE
txHandler: make dedup working set independent from ERL

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Fine tuning ERL might affect the dedup working set size in a negative way. This PR fixes that.
Additionally, make dedup creation optional (only if enabled). This is not important but looks a right thing to do.
Plus couple non-related small comment/assert noted along the way.

## Test Plan

Existing tests

### CODE REVIEW

File: data/txHandler.go [hunk 1]
```diff
@@ -174,13 +174,20 @@ func MakeTxHandler(opts TxHandlerOpts) (*TxHandler, error) {
...
 		net:                   opts.Net,
-		msgCache:              makeSaltedCache(2 * txBacklogSize),
-		txCanonicalCache:      makeDigestCache(2 * txBacklogSize),
 		cacheConfig:           txHandlerConfig{opts.Config.TxFilterRawMsgEnabled(), opts.Config.TxFilterCanonicalEnabled()},
 		streamVerifierChan:    make(chan execpool.InputJob),
 		streamVerifierDropped: make(chan *verify.UnverifiedTxnSigJob),
 	}
 
+	// use defaultBacklogSize = approx number of txns in a full block as a parameter for the dedup cache size
+	defaultBacklogSize := config.GetDefaultLocal().TxBacklogSize
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/txHandler.go:183]: config.GetDefaultLocal().TxBacklogSize returns real default and opts.Config.TxBacklogSize returns actual value and they could be different (that's where the issue comes from)

File: ledger/catchpointtracker_test.go [hunk 2]
```diff
@@ -467,6 +467,7 @@ func TestReproducibleCatchpointLabels(t *testing.T) {
 		if (uint64(i) >= cfg.MaxAcctLookback) && (uint64(i)-cfg.MaxAcctLookback > protoParams.CatchpointLookback) && ((uint64(i)-cfg.MaxAcctLookback)%cfg.CatchpointInterval == 0) {
 			ml.trackers.waitAccountsWriting()
 			catchpointLabels[i] = ct.GetLastCatchpointLabel()
+			require.NotEmpty(t, catchpointLabels[i])
```
Comment (AlgoAxel) [inline ledger/catchpointtracker_test.go:470]: is this meant to be here?

Comment (algorandskiy) [reply ledger/catchpointtracker_test.go:470]: not related but why not

File: data/txHandler.go [hunk 3]
```diff
@@ -174,13 +167,19 @@ func MakeTxHandler(opts TxHandlerOpts) (*TxHandler, error) {
...
 		net:                   opts.Net,
-		msgCache:              makeSaltedCache(2 * txBacklogSize),
-		txCanonicalCache:      makeDigestCache(2 * txBacklogSize),
-		cacheConfig:           txHandlerConfig{opts.Config.TxFilterRawMsgEnabled(), opts.Config.TxFilterCanonicalEnabled()},
 		streamVerifierChan:    make(chan execpool.InputJob),
 		streamVerifierDropped: make(chan *verify.UnverifiedTxnSigJob),
 	}
 
+	// use defaultBacklogSize = approx number of txns in a full block as a parameter for the dedup cache size
+	defaultBacklogSize := config.GetDefaultLocal().TxBacklogSize
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply data/txHandler.go:175]: it is the backlog size when ERL is disabled

**Review by algonautshant — APPROVED** [review]
This PR is addressing a fallout from a different parameter change. This PR is fine.

I am worried that too many parameters are controlling the critical pipeline of txn verification, and this will have adverse effects on the overall performance. This is out of scope of this PR.

**Review by AlgoAxel — APPROVED** [review]

**Comment by algorandskiy** [comment]
Thank you folks for looking into this.

**Pull request merged**